protobuf/v1/nexus.proto

```diff
@@ -382,3 +385,29 @@ message ResumeRebuildRequest {
 message ResumeRebuildResponse {
   Nexus nexus = 1;
 }
+
+message NexusCreateSnapshotRequest {
+  string nexus_uuid = 1;     // uuid of the nexus
+  string entity_id = 2;      // entity associated with this snapshot
+  string txn_id = 3;         // transaction id of the snapshot
+  string snapshot_name = 4;  // name of the snapshot
+  repeated NexusCreateSnapshotReplicaDescriptor replicas = 5; // replicas that participate in snapshot operation
+}
+
+message NexusCreateSnapshotReplicaDescriptor {
+  string replica_uuid = 1;           // uuid of the replica
+  optional string snapshot_uuid = 2; // uuid of the snapshot (can be omitted if replica is skipped)
+  bool skip = 3;                     // skip this replica
+}
+
+message NexusCreateSnapshotResponse {
```

Comment:
As per csi we can expose `size_bytes` of the snapshot.
For thick volumes to we take thin or thick snapshots? If they're thick then we probably do have a size for the snapshot?